packages/@lwc/engine-core/src/framework/restrictions.ts

```diff
@@ -305,14 +305,6 @@ function getCustomElementRestrictionsDescriptors(elm: HTMLElement): PropertyDesc
                     return appendChild.call(this, aChild);
                 },
             }),
-            removeChild: generateDataDescriptor({
-                value(this: Node, aChild: Node) {
-                    if (!isDomMutationAllowed) {
-                        reportInvalidSlotMutationInLightDOM(vm);
-                    }
-                    return removeChild.call(this, aChild);
-                },
-            }),
```

Comment:
In order to get this to work, I had to remove the `removeChild` restriction, because when resetting the root, we call `renderer.remove()`, which ultimately calls `parent.remove(child)`, which runs into this error.

I figure we're getting rid of this restriction anyway in https://github.com/salesforce/lwc/pull/2317. Plus I couldn't think of a straightforward workaround.

Reply:
Removing userland restrictions to get work done within the framework seems like a problem. Should we instead be invoking original DOM manipulation methods?

Reply:
@ekashida I considered doing that, yeah. But it seemed like overkill given that these restrictions are going away in #2317 anyway.

Also, I would have to modify `renderer.ts` to invoke the native `removeChild` instead, but none of the other parts of `renderer.ts` were doing that. Should `renderer.ts` be refactored to only use native DOM methods?

Reply:
I saw your comment regarding the slotting PR but I haven't spent enough time in there to understand why we would be removing restrictions for that feature.

> Should renderer.ts be refactored to only use native DOM methods?

Yes, I believe these methods are only used by the framework so it makes sense to do that. @pmdartus?

Reply:
@ekashida I just tried it out, and it looks like we actually can't do this in `renderer.ts`:

```js
const { removeChild } = Element.prototype;
/* ... */
remove(node: Node, parent: Node): void {
  removeChild.call(parent, node);
},
```

The reason is that we still need `removeChild` to go through the synthetic shadow patches. At least, we see errors like this one in the karma tests, when the parent is a document fragment:

![Screen Shot 2021-05-20 at 3 17 32 PM](https://user-images.githubusercontent.com/283842/119055850-931b7480-b97e-11eb-83c3-64f34fda40e6.png)